torch_topological/nn/multi_scale_kernel.py

```diff
@@ -0,0 +1,138 @@
+import torch
+
+from torch_topological.utils import wrap_if_not_iterable
+
+
+class MultiScaleKernel(torch.nn.Module):
+    # TODO: more detailed description
+    r"""Implement the multi-scale kernel between two persistence diagrams
+
+    This class implements the multi-scale kernel between two persistence
+    diagrams as defined by Reininghaus et al. as
+
+    .. math::
+        k_\sigma(F,G) = \frac{1}{8 \pi \sigma} \sum_{\substack{p \in F\\q \in G}} exp{-\frac{\|p-q\|^2}{8\sigma}} - exp{-\frac{\|p-\overline{q}\|^2}{8\sigma}}
+
+    where :math:`z=(z_1, z_2)` and :math:`\overline{z}=(z_2, z_1)`
+
+    Attributes
+    ----------
+    sigma : float
+        Scale parameter
+
+    Methods
+    -------
+    forward(X, Y)
+        Compute the kernel value between two persistence diagrams
+
+    References
+    ----------
+    .. [Reininghaus15] J. Reininghaus, U. Bauer and R. Kwitt, "A Stable
+       Multi-Scale Kernel for Topological Machine Learning", *Proceedings
+       of the IEEE conference on computer vision and pattern recognition*,
+       pp. 4741--4748, 2015.
+    """
+    def __init__(self, sigma):
+        """Create a new MultiScaleKernel instance
+
+        Parameters
+        ----------
+        sigma : float
+            scale parameter of the kernel
+        """
+        super().__init__()
+
+        self.sigma = sigma
+
+    @staticmethod
+    def _check_upper(d):
+        """
```

Comment:
Documentation should follow PEP-8 convention by first providing a brief one-sentence summary, followed by a more detailed description. For such a helper function, I would suggest that a multi-line comment before the function is perfectly sufficient.

Reply:
I'm not entirely sure if I got your suggestion right. Let me know if you would like to have a different style than the way I updated.

Reply:
This is perfect, thanks so much! Will resolve this and the other changes.